source/geh_common/src/geh_common/infrastructure/write_csv.py

```diff
@@ -184,6 +190,51 @@ def _get_file_info(
                 temporary=Path(tmpdir) / file_name,
             )
         )
+
+    if rows_per_file is not None and rows_per_file > 0:
+        _rename_single_chunk_filenames(file_info)
+
+    return file_info
+
+
+def _rename_single_chunk_filenames(file_info: list[FileInfo]) -> list[FileInfo]:
+    """Rename files where only one chunk exists in their partition by removing the chunk suffix.
+
+    Args:
+        file_info: A list of FileInfo objects containing file information.
+
+    Returns:
+        The updated list of FileInfo objects with renamed destination and temporary paths
+        for single-chunk partitions.
+    """
+    # Dict for partition_name and the number of chunks
+    files_by_partition = {}
+
+    # Dict for partition_name and the corresponding FileInfo object
+    partition_name_and_file_info = {}
+
+    # Go through all files and find the partitions and their chunks
+    for file in file_info:
+        name = file.destination.name
+        # Extract the partition name (everything before the last underscore)
+        partition_name = name[: name.rfind("_")]  # rfind() method finds the last occurrence of the specified value
```

Comment:
is it really the partition name? I guess it could be any string value depending on the callback?